### PR TITLE
Interpolation: Slightly change linear_interpolation() so that it can also interpolate vectors

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

* Generalize the wording in the documentation
* Don't initialize the result with `0`, but with the first term of a sum computed in a loop
* Add an example

## Release Management

* Affected package(s): Interpolation
* License and copyright ownership: stays at inria

### CODE REVIEW

File: Interpolation/include/CGAL/interpolation_functions.h [hunk 1]
```diff
@@ -73,13 +72,14 @@ linear_interpolation(ForwardIterator first, ForwardIterator beyond,
   typedef typename boost::result_of<ValueFunctor(arg_type)>::type                 result_type;
   typedef typename result_type::first_type                                        Value_type;
 
-  Value_type result(0);
-  result_type val;
+  result_type val = value_function(first->first);
```
Comment (MaelRL) [inline Interpolation/include/CGAL/interpolation_functions.h]: Need to check that the range is not empty somewhere.

File: Kernel_23/include/CGAL/Vector_3.h [hunk 2]
```diff
@@ -370,7 +370,7 @@ operator>>(std::istream& is, Vector_3<R>& v)
 {
   return extract(is, v, typename R::Kernel_tag() );
 }
-
+
```
Comment (MaelRL) [inline Kernel_23/include/CGAL/Vector_3.h]: some bad whitespace changes throughout the diff.

File: Interpolation/include/CGAL/interpolation_functions.h (hunk 1 above)
Comment (afabri) [reply Interpolation/include/CGAL/interpolation_functions.h]: I added  preconditions for all interpolation functions in the code and the manual. 
@lrineau: Note that this is not a change as we already now have the precondition that the sum of weights `norm` is larger than zero, which implies a non-empty range.

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.14-Ic-27][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-27.shtml

**Pull request merged**